.github/workflows/code-quality.yaml

```diff
@@ -0,0 +1,42 @@
+name: Code Quality Checks
+on:
+  push:
+    branches:
+    - main
+    - release/**
+  pull_request:
+    branches:
+    - main
+    - release/**
+  workflow_call:
+  workflow_dispatch:
+# Cancel old runs when a new commit is pushed to the same branch if not on main or dev
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+defaults:
+  run:
+    working-directory: .
+jobs:
+  code-quality:
+    runs-on: ubuntu-latest  # todo: switch to linux-ubuntu-latest later
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        python_version:
+        - "3.9"
+        - "3.10"
```

Comment:
pretty sure this is inconsistent

Reply:
[composer](https://github.com/mosaicml/composer/blob/358fbeb0c961eb206a183305554c2fda38488232/.github/workflows/code-quality.yaml#L24) tests against python 3.9, 3.10, 3.11. I'll add tests for 3.11. Is this what you meant about consistency?

Reply:
Depends on what python versions you want to support. @mihir-db ? I meant your gpu test was 3.11 only and this had 3.9 and 3.10 only.

Reply:
Composer supports python 3.9, 3.10, 3.11 for their code-quality tests. We have this here.

Composer supports python 3.9 or 3.11 for their gpu-pr tests. In their yaml, the [container](https://github.com/mosaicml/composer/blob/a15b18ce18db6df4cff4e10251adcb9d5c5845db/.github/workflows/pr-gpu.yaml#L17) has python 3.11 but the [python-version](https://github.com/mosaicml/composer/blob/a15b18ce18db6df4cff4e10251adcb9d5c5845db/.github/workflows/pr-gpu.yaml#L32C7-L32C21) is 3.9. I chose to use python 3.11 because it is more up to date.

I do think it may be a bit repetitive/not the best use of GPU resources to run the gpu tests on multiple versions of python. However, I'm happy to add these tests if they are deemed necessary.

Reply:
GPU tests should only be one python version whereas codeql is rotating 3 as we have in Composer. I think this is correct.

Reply:
As per in-person discussion, we are only supporting python 3.11 and GA's no longer running on 3.9 or 3.10.